### PR TITLE
Gorm refactor

### PR DESCRIPTION
**Pull request by ericharmeling**
Fixes https://github.com/cockroachdb/docs/issues/4207
Fixes https://github.com/cockroachdb/docs/issues/8683
Fixes https://github.com/cockroachdb/docs/issues/8228

This PR updates the existing GORM tutorial to match the refactored sample app now located at https://github.com/cockroachlabs/example-app-go-gorm. It also includes a sample tutorial for the new Hello World app located at https://github.com/cockroachlabs/hello-world-go-gorm.

### CODE REVIEW

File: v20.2/build-a-go-app-with-cockroachdb-gorm.md [hunk 1]
```diff
@@ -13,92 +13,117 @@ referral_id: docs_hello_world_go_gorm
...
-
-The following code uses the [GORM](http://gorm.io) ORM (v1) to map Go-specific objects to SQL operations, and the [`crdbgorm`](https://godoc.org/github.com/cockroachdb/cockroach-go/crdb/crdbgorm) package to handle [transactions](transactions.html). Specifically:
-
-- `db.AutoMigrate(&Account{})` creates an `accounts` table based on the Account model.
-- `db.Create(&Account{})` inserts rows into the table.
-- `db.Find(&accounts)` selects from the table so that balances can be printed.
-- The funds transfer occurs in `transferFunds()`. To ensure that we [handle retry errors](transactions.html#client-side-intervention), we wrap the function call in [`crdbgorm.ExecuteTx()`](https://github.com/cockroachdb/cockroach-go/blob/master/crdb/crdbgorm/gorm.go#L29).
+{% include_cached copy-clipboard.html %}
+~~~ shell
+$ git clone https://github.com/cockroachlabs/example-app-go-gorm/
```
Comment (ianjevans) [inline v20.2/build-a-go-app-with-cockroachdb-gorm.md:32]:
```suggestion
$ git clone https://github.com/cockroachlabs/example-app-go-gorm
```

File: v20.2/hello-world-go-gorm.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+{% include {{page.version.version}}/app/sample-setup.md %}
+
+## Step 2. Get the code
+
+Clone the code's GitHub repo:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+$ git clone https://github.com/cockroachlabs/hello-world-go-gorm/
```
Comment (ianjevans) [inline v20.2/hello-world-go-gorm.md:26]:
```suggestion
$ git clone https://github.com/cockroachlabs/hello-world-go-gorm
```

File: v21.1/build-a-go-app-with-cockroachdb-gorm.md [hunk 3]
```diff
@@ -13,92 +13,117 @@ referral_id: docs_hello_world_go_gorm
...
-
-The following code uses the [GORM](http://gorm.io) ORM (v1) to map Go-specific objects to SQL operations, and the [`crdbgorm`](https://godoc.org/github.com/cockroachdb/cockroach-go/crdb/crdbgorm) package to handle [transactions](transactions.html). Specifically:
-
-- `db.AutoMigrate(&Account{})` creates an `accounts` table based on the Account model.
-- `db.Create(&Account{})` inserts rows into the table.
-- `db.Find(&accounts)` selects from the table so that balances can be printed.
-- The funds transfer occurs in `transferFunds()`. To ensure that we [handle retry errors](transactions.html#client-side-intervention), we wrap the function call in [`crdbgorm.ExecuteTx()`](https://github.com/cockroachdb/cockroach-go/blob/master/crdb/crdbgorm/gorm.go#L29).
+{% include_cached copy-clipboard.html %}
+~~~ shell
+$ git clone https://github.com/cockroachlabs/example-app-go-gorm/
```
Comment (ianjevans) [inline v21.1/build-a-go-app-with-cockroachdb-gorm.md:32]:
```suggestion
$ git clone https://github.com/cockroachlabs/example-app-go-gorm
```

File: v21.1/hello-world-go-gorm.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+{% include {{page.version.version}}/app/sample-setup.md %}
+
+## Step 2. Get the code
+
+Clone the code's GitHub repo:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+$ git clone https://github.com/cockroachlabs/hello-world-go-gorm/
```
Comment (ianjevans) [inline v21.1/hello-world-go-gorm.md:26]:
```suggestion
$ git clone https://github.com/cockroachlabs/hello-world-go-gorm
```

File: v21.2/build-a-go-app-with-cockroachdb-gorm.md [hunk 5]
```diff
@@ -13,92 +13,117 @@ referral_id: docs_hello_world_go_gorm
...
-
-The following code uses the [GORM](http://gorm.io) ORM (v1) to map Go-specific objects to SQL operations, and the [`crdbgorm`](https://godoc.org/github.com/cockroachdb/cockroach-go/crdb/crdbgorm) package to handle [transactions](transactions.html). Specifically:
-
-- `db.AutoMigrate(&Account{})` creates an `accounts` table based on the Account model.
-- `db.Create(&Account{})` inserts rows into the table.
-- `db.Find(&accounts)` selects from the table so that balances can be printed.
-- The funds transfer occurs in `transferFunds()`. To ensure that we [handle retry errors](transactions.html#client-side-intervention), we wrap the function call in [`crdbgorm.ExecuteTx()`](https://github.com/cockroachdb/cockroach-go/blob/master/crdb/crdbgorm/gorm.go#L29).
+{% include_cached copy-clipboard.html %}
+~~~ shell
+$ git clone https://github.com/cockroachlabs/example-app-go-gorm/
```
Comment (ianjevans) [inline v21.2/build-a-go-app-with-cockroachdb-gorm.md:32]:
```suggestion
$ git clone https://github.com/cockroachlabs/example-app-go-gorm
```

File: v21.2/hello-world-go-gorm.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+
+{% include {{page.version.version}}/app/sample-setup.md %}
+
+## Step 2. Get the code
+
+Clone the code's GitHub repo:
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+$ git clone https://github.com/cockroachlabs/hello-world-go-gorm/
```
Comment (ianjevans) [inline v21.2/hello-world-go-gorm.md:26]:
```suggestion
$ git clone https://github.com/cockroachlabs/hello-world-go-gorm
```

**Review by ianjevans — CHANGES_REQUESTED** [review]
The `git clone` steps need the trailing slash removed, as it will error-out. Otherwise looks good.

**Review by ericharmeling — COMMENTED** [review]
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/11704#-:-Mk2We6w9R7Lg7u0u2j-:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @ianjevans)

___
*[v20.2/build-a-go-app-with-cockroachdb-gorm.md, line 32 at r1](https://reviewable.io/reviews/cockroachdb/docs/11704#-MjkSeHHc1r5PEqlQmT0-r1-32:-Mk2WfMY37vReXEQN7Bm:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/d513572d8b643c8fac6b45f2cb14028b8a734afd/v20.2/build-a-go-app-with-cockroachdb-gorm.md#L32)):*
<details><summary><i>Previously, ianjevans (Ian Evans) wrote…</i></summary><blockquote>

```suggestion
$ git clone https://github.com/cockroachlabs/example-app-go-gorm
```
</blockquote></details>

Done.
___
*[v20.2/hello-world-go-gorm.md, line 26 at r1](https://reviewable.io/reviews/cockroachdb/docs/11704#-MjkSeHHc1r5PEqlQmT2-r1-26:-Mk2Wg5a7K8a5knnExDp:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/d513572d8b643c8fac6b45f2cb14028b8a734afd/v20.2/hello-world-go-gorm.md#L26)):*
<details><summary><i>Previously, ianjevans (Ian Evans) wrote…</i></summary><blockquote>

```suggestion
$ git clone https://github.com/cockroachlabs/hello-world-go-gorm
```
</blockquote></details>

Done.
___
*[v21.1/build-a-go-app-with-cockroachdb-gorm.md, line 32 at r1](https://reviewable.io/reviews/cockroachdb/docs/11704#-MjkSeHHc1r5PEqlQmT4-r1-32:-Mk2WgYc37reEnOAy4-4:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/d513572d8b643c8fac6b45f2cb14028b8a734afd/v21.1/build-a-go-app-with-cockroachdb-gorm.md#L32)):*
<details><summary><i>Previously, ianjevans (Ian Evans) wrote…</i></summary><blockquote>

```suggestion
$ git clone https://github.com/cockroachlabs/example-app-go-gorm
```
</blockquote></details>

Done.
___
*[v21.1/hello-world-go-gorm.md, line 26 at r1](https://reviewable.io/reviews/cockroachdb/docs/11704#-MjkSeHHc1r5PEqlQmT6-r1-26:-Mk2WgwwC3PLcOGAqb0A:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/d513572d8b643c8fac6b45f2cb14028b8a734afd/v21.1/hello-world-go-gorm.md#L26)):*
<details><summary><i>Previously, ianjevans (Ian Evans) wrote…</i></summary><blockquote>

```suggestion
$ git clone https://github.com/cockroachlabs/hello-world-go-gorm
```
</blockquote></details>

Done.
___
*[v21.2/build-a-go-app-with-cockroachdb-gorm.md, line 32 at r1](https://reviewable.io/reviews/cockroachdb/docs/11704#-MjkSeHHc1r5PEqlQmT8-r1-32:-Mk2WhES6Do_3F03-J_1:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/d513572d8b643c8fac6b45f2cb14028b8a734afd/v21.2/build-a-go-app-with-cockroachdb-gorm.md#L32)):*
<details><summary><i>Previously, ianjevans (Ian Evans) wrote…</i></summary><blockquote>

```suggestion
$ git clone https://github.com/cockroachlabs/example-app-go-gorm
```
</blockquote></details>

Done.
___
*[v21.2/hello-world-go-gorm.md, line 26 at r1](https://reviewable.io/reviews/cockroachdb/docs/11704#-MjkSeHHc1r5PEqlQmTA-r1-26:-Mk2WhsS5JMqKJWddvdh:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/d513572d8b643c8fac6b45f2cb14028b8a734afd/v21.2/hello-world-go-gorm.md#L26)):*
<details><summary><i>Previously, ianjevans (Ian Evans) wrote…</i></summary><blockquote>

```suggestion
$ git clone https://github.com/cockroachlabs/hello-world-go-gorm
```
</blockquote></details>

Done.

**Pull request merged**